pkg/core/resources/model/resource.go

```diff
@@ -704,6 +704,26 @@ func TargetRefToResourceIdentifier(meta ResourceMeta, tr common_api.TargetRef) R
 	}
 }
 
+func ResolveBackendRef(meta ResourceMeta, br common_api.BackendRef) ResolvedBackendRef {
+	resolved := ResolvedBackendRef{LegacyBackendRef: &br}
+
+	switch {
+	case br.Kind == common_api.MeshService && br.ReferencesRealObject():
+	case br.Kind == common_api.MeshExternalService:
+	case br.Kind == common_api.MeshMultiZoneService:
+	default:
+		return resolved
+	}
+
+	resolved.Resource = &TypedResourceIdentifier{
+		ResourceIdentifier: TargetRefToResourceIdentifier(meta, br.TargetRef),
+		ResourceType:       ResourceType(br.Kind),
+		SectionName:        fmt.Sprintf("%d", br.Port),
```

Comment:
We should check if the port is `nil`. I think it's not required for `MeshExternalService`, which might cause a panic.

Reply:
Good catch, fixed